src/pages/Lend/LendByCategoryPage.vue

```diff
@@ -398,30 +445,53 @@ export default {
 			}
 		},
 		initializeRecommendedLoansRow() {
-			const recLoanChannel = _find(this.categorySetting, { __typename: 'RecLoanChannel' });
-			if (recLoanChannel) {
+			const recLoanChannels = this.categorySetting.filter(setting => {
+				// eslint-disable-next-line no-underscore-dangle
+				return setting.__typename === 'RecLoanChannel';
+			});
+			console.log('category setting', this.categorySetting);
+			console.log('recLoanChannels', recLoanChannels);
+
+			if (recLoanChannels.length) {
 				// Load recommended loan data
 				try {
 					const variables = {
 						basketId: cookieStore.get('kvbskt'),
-						ids: [recLoanChannel.id],
+						ids: recLoanChannels.map(channel => channel.id),
 						imgDefaultSize: this.showHoverLoanCards ? 'w480h300' : 'w480h360',
 						imgRetinaSize: this.showHoverLoanCards ? 'w960h600' : 'w960h720',
 					};
 					const data = this.apollo.readQuery({
 						query: recommendedLoansQuery,
 						variables,
 					});
+					console.log('data after read query', data);
+
+					const allRecLoanChannels = data.ml?.getOrderedChannelsByIds ?? [];
+
+					const allRecLoans = allRecLoanChannels.map(channel => {
+						const channelLoans = channel.loans?.values ?? [];
```

Comment:
You'll want to create another const here for to extract data from the rec channel stored in categorySetting.

`const channelData = this.categorySetting.filter(cat => id === channel.id);`

Something like this should give you an object of the original channel data including it's name and description. You can then use these in the object you're build below where appropriate.

Reply:
catrgorySetting only has channel ids, so I don't think it can be used like that.

Reply:
You're right! we worked through that directly and used data from recommended channels object